engine/table/src/main/java/io/deephaven/engine/table/impl/select/PatternFindFilter.java

```diff
@@ -0,0 +1,209 @@
+ *
+ * @see PatternMatchesFilter
+ */
+public final class PatternFindFilter extends WhereFilterImpl implements ObjectChunkFilter<String> {
+
+    private static final long serialVersionUID = 1L;
+
+    public static PatternFindFilter stringContainsFilter(
+            MatchType matchType,
+            String columnName,
+            String... values) {
+        return stringContainsFilter(CaseSensitivity.MatchCase, matchType, columnName, values);
+    }
+
+    public static PatternFindFilter stringContainsFilter(
+            String columnName,
+            String... values) {
+        return stringContainsFilter(MatchType.Regular, columnName, values);
+    }
+
+    public static PatternFindFilter stringContainsFilter(
+            CaseSensitivity sensitivity,
+            MatchType matchType,
+            @NotNull String columnName,
+            String... values) {
+        return stringContainsFilter(sensitivity, matchType, columnName, true, false, values);
+    }
+
+    public static PatternFindFilter stringContainsFilter(
+            CaseSensitivity sensitivity,
+            MatchType matchType,
+            @NotNull String columnName,
+            boolean internalDisjuntive,
+            boolean removeQuotes,
+            String... values) {
+        final String value = constructRegex(values, matchType, internalDisjuntive, removeQuotes, columnName);
+        // note: this is persisting old behavior w/ matchNulls = false
+        return new PatternFindFilter(
+                ColumnName.of(columnName),
+                Pattern.compile(value, sensitivity == CaseSensitivity.IgnoreCase ? Pattern.CASE_INSENSITIVE : 0),
+                matchType == MatchType.Inverted,
+                false);
+    }
+
+    private final ColumnName columnName;
+    private final Pattern pattern;
+    private final boolean invertPattern;
+    private final boolean matchNulls;
+
+    public PatternFindFilter(ColumnName columnName, Pattern pattern, boolean invertPattern, boolean matchNulls) {
+        this.pattern = Objects.requireNonNull(pattern);
+        this.columnName = Objects.requireNonNull(columnName);
+        this.invertPattern = invertPattern;
+        this.matchNulls = matchNulls;
+    }
+
+    @Override
+    public void init(TableDefinition tableDefinition) {
+        final ColumnDefinition<?> column = tableDefinition.getColumn(columnName.name());
+        if (column == null) {
+            throw new RuntimeException(String.format("Column '%s' doesn't exist in this table, available columns: %s",
+                    columnName.name(), tableDefinition.getColumnNames()));
+        }
+    }
+
+    @Override
+    public WritableRowSet filter(RowSet selection, RowSet fullSet, Table table, boolean usePrev) {
+        final ColumnSource<String> columnSource = table.getColumnSource(columnName.name());
+        return ChunkFilter.applyChunkFilter(selection, columnSource, usePrev, this);
```

Comment:
We think we want to:
1. Cleanup references to `StringContainsFilter`
2. Rename `constructRegex` to say what it does
3. Combine these classes, and just have different chunk filter implementations.